src/main.rs

```diff
@@ -737,6 +766,48 @@ fn backtrace(
             false
         };
 
+        // This is our first run through the loop, some initial handling and printing is required
+        // TODO refactor this
+        if frame_index == 0 {
+            if pc & !THUMB_BIT == vector_table.hard_fault & !THUMB_BIT {
```

Comment:
that's a pretty deep & long nesting, I'd extract into function

Reply:
totally– that's why I put a // TODO refactor this in there ;) My secret plan was to do this as a drive-by when I tackle the next issue. Didn't want to overhaul everything in one pass because then

my functionality changes would get lost, making it harder to review
time budget is limited